test/check/check.go

```diff
@@ -120,6 +122,36 @@ func PlacementListFromComponentDaemonSets(component Component) ([]cnao.Placement
 	return placementList, nil
 }
 
+func getMonitoringNamespace() (string, error) {
+	envVars, err := GetEnvVarsFromDeployment(components.Name)
+	if err != nil {
+		return "", err
+	}
+	for _, envVar := range envVars {
+		if envVar.Name == "MONITORING_NAMESPACE" {
+			return envVar.Value, nil
+		}
+	}
+	return "", errors.New("MONITORING_NAMESPACE env variable not found on operator")
+}
+
+func PrometheusDeployedOnCluster() (bool, error) {
+	monitoringNamespace, err := getMonitoringNamespace()
+	if err != nil {
+		return false, err
+	}
+
+	ns := corev1.Namespace{}
+	err = framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: monitoringNamespace}, &ns)
+	if err == nil {
+		return true, nil
+	} else if apierrors.IsNotFound(err) {
+		return false, nil
+	}
+
+	return false, err
```

Comment:
We have to make this more idiomatic doing some kind of fail fast.

```golang
if err != nil {
  if apierrors.IsNotFound(err) {
    return false, nil
  }
  return false, err
}
return false, nil
```

Reply:
removed the function entirely in favor on the production function